BossMod/Util/ShapeDistance.cs

```diff
@@ -199,6 +199,6 @@ Func<WPos, float> edge((WPos p1, WPos p2) e)
     }
     public static Func<WPos, float> ConvexPolygon(IEnumerable<WPos> vertices, bool cw, float offset = 0) => ConvexPolygon(PolygonUtil.EnumerateEdges(vertices), cw, offset);
 
-    public static Func<WPos, float> Intersection(List<Func<WPos, float>> funcs, float offset = 0) => p => funcs.Max(e => e(p)) - offset;
-    public static Func<WPos, float> Union(List<Func<WPos, float>> funcs, float offset = 0) => p => funcs.Min(e => e(p)) - offset;
+    public static Func<WPos, float> Intersection(IEnumerable<Func<WPos, float>> funcs, float offset = 0) => p => funcs.Max(e => e(p)) - offset;
+    public static Func<WPos, float> Union(IEnumerable<Func<WPos, float>> funcs, float offset = 0) => p => funcs.Min(e => e(p)) - offset;
```

Comment:
This function is called multiple (thousands!) of times, so the reason it was a list is for performance (we don't want to enumerate arbitrarily complex enumerable multiple times, plus technically enumerable could even be single-pass).